### PR TITLE
Change the reference document link

### PR DESCRIPTION
**Pull request by NoCtrlZ**
Hi there!
I modified the document link and fixed some typos.
When I click the document link, there is a 404 error so I set the link about it.
But I don't know about the other bellow three documents link.

And when I ran `$make doc` command the error happened because of the toolchain version problem.
The error is following and the issue is [here](https://github.com/spruceid/didkit/issues/126).

```shell
(base) ➜  plonk git:(master) make doc
   Compiling syn v1.0.72
    Checking crossbeam-epoch v0.9.5
error[E0433]: failed to resolve: could not find `addr_of` in `ptr`
   --> /Users/user/.cargo/registry/src/github.com-1ecc6299db9ec823/crossbeam-epoch-0.9.5/src/internal.rs:620:59
    |
620 |         let local_ptr = (entry as *const Entry as usize - offset_of!(Local, entry)) as *const Local;
    |                                                           ^^^^^^^^^^^^^^^^^^^^^^^^ could not find `addr_of` in `ptr`
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `addr_of` in `ptr`
   --> /Users/user/.cargo/registry/src/github.com-1ecc6299db9ec823/crossbeam-epoch-0.9.5/src/internal.rs:613:59
    |
613 |         let entry_ptr = (local as *const Local as usize + offset_of!(Local, entry)) as *const Entry;
    |                                                           ^^^^^^^^^^^^^^^^^^^^^^^^ could not find `addr_of` in `ptr`
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0433`.
error: could not compile `crossbeam-epoch`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
make: *** [doc] Error 101
```

Please confirm.
Thank you!

### CODE REVIEW

File: rust-toolchain [hunk 1]
```diff
@@ -1 +1 @@
-nightly
+nightly-2021-03-16
```
Comment (CPerezz) [inline rust-toolchain:1]: Please, leave the toolchain as set in `master` or in case of collision, to `nightly-06-06-2021`.

**Review by CPerezz — CHANGES_REQUESTED** [review]
LGTM but the toolchain needs to be changed  to match https://github.com/dusk-network/plonk/blob/65e1a51c113603c23d3979e5367ee6832bba1bb2/rust-toolchain#L1

**Comment by CPerezz** [comment]
Also, please add the changes under `Unreleased` section in `CHANGELOG.md`

**Comment by NoCtrlZ** [comment]
Hi @CPerezz
Thank you for the review.
I fixed it according to your review.
Please confirm.
By the way, is there any developer community like discord?
I'd like to ask some questions about this implementation :sunglasses:
Thank you!

**Comment by NoCtrlZ** [comment]
I added one change.
When I compiled the modules, the error happened `dusk-bytes` specific version.
so I set `dusk-bytes` module version as `0.1.4`.
```
(base) ➜  plonk git:(master) cargo test
   Compiling libc v0.2.95
   Compiling proc-macro2 v1.0.27
   Compiling autocfg v1.0.1
   Compiling unicode-xid v0.2.2
   Compiling cfg-if v1.0.0
   Compiling syn v1.0.72
   Compiling crossbeam-utils v0.8.5
   Compiling lazy_static v1.4.0
   Compiling crossbeam-epoch v0.9.5
   Compiling rayon-core v1.9.1
   Compiling scopeguard v1.1.0
   Compiling either v1.6.1
   Compiling byteorder v1.4.3
   Compiling subtle v2.4.0
   Compiling ahash v0.4.7
   Compiling keccak v0.1.0
   Compiling remove_dir_all v0.5.3
   Compiling itertools v0.9.0
   Compiling hashbrown v0.9.1
   Compiling memoffset v0.6.4
   Compiling rayon v1.5.1
   Compiling crossbeam-channel v0.5.1
   Compiling getrandom v0.2.3
   Compiling num_cpus v1.13.0
   Compiling rand v0.4.6
   Compiling rand_core v0.6.2
   Compiling quote v1.0.9
   Compiling crossbeam-deque v0.8.0
   Compiling tempdir v0.3.7
   Compiling synstructure v0.12.4
   Compiling derive-hex v0.1.1
   Compiling zeroize_derive v1.1.0
   Compiling dusk-bytes v0.1.3
   Compiling zeroize v1.3.0
error[E0557]: feature has been removed
 --> /Users/user/.cargo/registry/src/github.com-1ecc6299db9ec823/dusk-bytes-0.1.3/src/lib.rs:7:12
  |
7 | #![feature(external_doc)]
  |            ^^^^^^^^^^^^ feature has been removed
  |
  = note: use #[doc = include_str!("filename")] instead, which handles macro invocations

error: aborting due to previous error

For more information about this error, try `rustc --explain E0557`.
error: could not compile `dusk-bytes`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by CPerezz** [comment]
Hi, all seems nice. I'd maybe ask for an edit of the commit message of e38fe5c to something like: `Update CHANGELOG.md`. Which follows the same format as the others.

For the rest, looks fine!


> By the way, is there any developer community like discord?
> I'd like to ask some questions about this implementation sunglasses
> Thank you!

You can use [Github Discussions](https://github.com/dusk-network/plonk/discussions)!! We don't have any other places that I'm aware.

**Review by CPerezz — APPROVED** [review]
LGTM

Please, address:
> Hi, all seems nice. I'd maybe ask for an edit of the commit message of e38fe5c to something like: Update CHANGELOG.md. Which follows the same format as the others.


Once addressed, this can be merged!!
Thanks for your contribution! :smile:

**Comment by NoCtrlZ** [comment]
Hi @CPerezz 
Thank you for the review!
I changed the commit message according to your review.
Please confirm!

**Pull request merged**